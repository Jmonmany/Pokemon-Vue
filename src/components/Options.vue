<template>
  <div class="options-container">
    <ul>
      <li v-for="pokemon in pokeOptions" :key="pokemon.id" @click="$emit('selection', pokemon.id)">
        {{ pokemon.name }}
      </li>
      <!-- the $emit method is used to emit an event to the parent component -->
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    pokeOptions: {
      type: Array,
      required: true,
    },
  },
};
</script>

<style>
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
