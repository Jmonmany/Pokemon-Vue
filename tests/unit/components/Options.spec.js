import { shallowMount } from "@vue/test-utils";
import Options from "@/components/Options";
import { pokemonsMock } from "../mocks/pokemons.mock.js";
describe("Options.vue component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(Options, {
      props: {
        pokeOptions: pokemonsMock,
      },
    });
  });

  test("should match with snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("should render the 4 options correctly", () => {
    const options = wrapper.findAll("li");
    expect(options.length).toBe(4);
    expect(options[0].text()).toBe("bulbasaur");
    expect(options[1].text()).toBe("ivysaur");
    expect(options[2].text()).toBe("venusaur");
    expect(options[3].text()).toBe("charmander");
  });
  test("should emit an event with the correct payload", () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");
    li1.trigger("click");
    expect(wrapper.emitted("selection")[0]).toEqual([1]);
    li2.trigger("click");
    expect(wrapper.emitted("selection")[1]).toEqual([2]);
    li3.trigger("click");
    expect(wrapper.emitted("selection")[2]).toEqual([3]);
    li4.trigger("click");
    expect(wrapper.emitted("selection")[3]).toEqual([4]);
  });
});
