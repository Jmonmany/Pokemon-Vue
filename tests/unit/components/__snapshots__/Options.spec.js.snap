// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options.vue component should match with snapshot 1`] = `
<div class="options-container">
  <ul>
    <li>bulbasaur</li>
    <li>ivysaur</li>
    <li>venusaur</li>
    <li>charmander</li><!-- the $emit method is used to emit an event to the parent component -->
  </ul>
</div>
`;
